### PR TITLE
[Github actions] adding test github actions and adds test changes for pytest

### PR DESCRIPTION
**Pull request by Harveyt47**
Creating test.yml. Still need to add test changes for this to work.

### CODE REVIEW

**Comment by Harveyt47** [comment]
@mmcky I still need to add in coveralls

**Comment by mmcky** [comment]
thanks @Harveyt47 -- if you can give me an update on this sometime that would be great. Thanks.

**Comment by Harveyt47** [comment]
Hey @mmcky look like I've got coveralls working and `pytest` passing. I'm not sure why the coverage is so low though

**Comment by mmcky** [comment]
Perhaps the `.coveragerc` file is not being used?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest tables
+        pip install pytest-cov
+        python setup.py develop
+        pip install -U pip nose coverage numpy scipy pandas numba sympy ipython statsmodels flake8
```
Comment (mmcky) [inline .github/workflows/test.yml:24]: @Harveyt47 you can remove `nose` here right?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        pip install pytest tables
+        pip install pytest-cov
+        python setup.py develop
+        pip install -U pip nose coverage numpy scipy pandas numba sympy ipython statsmodels flake8 
+        python setup.py install
+        pip install coveralls
+    - name: Run Tests
+      
+      env:
+          COVERALLS_REPO_TOKEN: 7f8JDNk6BsPMESy0URL3W00zlazzRcuzR
```
Comment (mmcky) [inline .github/workflows/test.yml:30]: Can you remind me what this REPO token is for?

Is this for `travis` to post the `coverage` results back to the REPO (or something else)?

We may need to rework this part for `GitHub Actions`.

**Comment by mmcky** [comment]
@Harveyt47 this may be your issue re: coverage stats. 

```
PyYAML is not installed, skipping .coveralls.yml.
Submitting coverage to coveralls.io...
Coverage submitted!
Job #2e367460bc5316537ebbba8da13cf8f9b92d4ed4.1
https://coveralls.io/jobs/63716041
```

so the token must be for this service. 

This means we should remove it and add it as a secret. 

``` 
${{ secrets. COVERALLS_REPO_TOKEN }}
```

**Comment by mmcky** [comment]
@Harveyt47 it looks like this is running the tests twice due to use of `pull_request` and `push`

Best to set this up for `pull_request` and then a separate `on:` for `push` to `- branch: master`

**Comment by mmcky** [comment]
thanks @Harveyt47 is there a way to get the Coveralls details as a comment on the PR?

**Comment by Harveyt47** [comment]
Hi @mmcky I tried changing the Coveralls set up as I believe it thinks it using `travis-ci` as the build type, as you can see when you click the [link](https://coveralls.io/builds/31382904) in the coveralls comment above. I think that's why the coverage was so low. I have used a new set up which is from the coveralls `Github Actions` marketplace page [here]( https://github.com/marketplace/actions/coveralls-github-action)

**Comment by mmcky** [comment]
@Harveyt47 maybe we should remove the `yml` file now that #561 has been merged and change this to be the PR to update tests for `pytest`

**Comment by oyamad** [comment]
Can this be closed?

**Comment by mmcky** [comment]
I agree -- thanks @oyamad

**Pull request closed without merging**